hugr-core/src/export.rs

```diff
@@ -392,6 +406,58 @@ impl<'a> Context<'a> {
         node_id
     }
 
+    /// Export an `OpDef` as an operation declaration.
+    ///
+    /// Operations that allow a declarative form are exported as a reference to
+    /// an operation declaration node, and this node is reused for all instances
+    /// of the operation. The node is added to the `decl_operations` map so that
+    /// at the end of the export, the operation declaration nodes can be added
+    /// to the module as children of the module region.
+    pub fn export_opdef(&mut self, opdef: &OpDef) -> model::GlobalRef<'a> {
+        use std::collections::hash_map::Entry;
+
+        let poly_func_type = match opdef.signature_func() {
+            SignatureFunc::PolyFuncType(poly_func_type) => poly_func_type,
+            _ => return self.make_named_global_ref(opdef.extension(), opdef.name()),
+        };
+
+        let key = (opdef.extension().clone(), opdef.name().clone());
+        let entry = self.decl_operations.entry(key);
+
+        let node = match entry {
+            Entry::Occupied(occupied_entry) => {
+                return model::GlobalRef::Direct(*occupied_entry.get())
+            }
+            Entry::Vacant(vacant_entry) => {
+                *vacant_entry.insert(self.module.insert_node(model::Node {
+                    operation: model::Operation::Invalid,
+                    inputs: &[],
+                    outputs: &[],
+                    params: &[],
+                    regions: &[],
+                    meta: &[], // TODO: Metadata
+                    signature: None,
+                }))
+            }
+        };
+
+        let decl = self.with_local_scope(node, |this| {
```

Comment:
is this going to overwrite the polyfunctype even if the node is already occupied?

Reply:
I don't understand what you are referring to; can you rephrase that?

Reply:
IIUC this function looks up whether an opdef with the same name already exists in the map. On line 455 it appears the `operation` field of that node is overwritten by the declaration whether or not it was already occupied by a previous declaration. Is this intended behaviour?

Reply:
If the name already exists in the map, the node for that name is returned immediately. If it does not exist, we create a placeholder entry first and then overwrite that placeholder entry on line 455. No previous declaration will be overwritten.